flexmeasures_openweathermap/utils/locating.py

```diff
@@ -105,3 +106,15 @@ def find_weather_sensor_by_location_or_fail(
             % sensor_name
         )
     return weather_sensor
+
+
+def get_asset_id_location(asset_id: int) -> List[Tuple[float, float]]:
+    """Get location for forecasting by passing an asset id"""
+    asset = GenericAsset.query.filter(
+        GenericAsset.generic_asset_type_id == asset_id
+    ).one_or_none()
+    if asset is None:
+        raise Exception(
+            "[FLEXMEASURES-OWM] No asset found for the given asset id %s." % asset_id
+        )
```

Comment:
Also raise an Exception if an asset is found, but it's `generic_asset_type.name != locating.WEATHER_STATION_TYPE_NAME`